diff -Naur a/e2fsck/Makefile.in b/e2fsck/Makefile.in
--- a/e2fsck/Makefile.in	2016-01-20 19:48:04.968090847 +0100
+++ b/e2fsck/Makefile.in	2016-01-20 19:49:00.028086336 +0100
@@ -47,7 +47,7 @@
 #
 #MCHECK=	-checker
 #LIBS= $(LIBCOM_ERR) $(LIB_EXT2FS) $(CHECKLIB)
-#DEPLIBS= $(LIBCOM_ERR) $(LIB_EXT2FS) 
+#DEPLIBS= $(LIBCOM_ERR) $(LIB_EXT2FS)
 #CHECKLIB= /usr/lib/libchecker.o
 
 #
@@ -124,16 +124,16 @@
 
 e2fsck: $(OBJS)  $(DEPLIBS)
 	$(E) "	LD $@"
-	$(Q) $(LD) $(ALL_LDFLAGS) $(RDYNAMIC) -o e2fsck $(OBJS) $(LIBS) 
+	$(Q) $(LD) $(ALL_LDFLAGS) $(RDYNAMIC) -o e2fsck $(OBJS) $(LIBS)
 
 e2fsck.static: $(OBJS) $(STATIC_DEPLIBS)
 	$(E) "	LD $@"
-	$(Q) $(LD) $(LDFLAGS_STATIC) -o e2fsck.static $(OBJS) $(STATIC_LIBS) 
+	$(Q) $(LD) $(LDFLAGS_STATIC) -o e2fsck.static $(OBJS) $(STATIC_LIBS)
 
 e2fsck.profiled: $(OBJS)  $(PROFILED_DEPLIBS)
 	$(E) "	LD $@"
 	$(Q) $(LD) $(ALL_LDFLAGS) -g -pg -o e2fsck.profiled $(PROFILED_OBJS) \
-		$(PROFILED_LIBS) 
+		$(PROFILED_LIBS)
 
 gen_crc32table: $(srcdir)/gen_crc32table.c
 	$(E) "	CC $@"
@@ -162,7 +162,7 @@
 tst_refcount: ea_refcount.c $(DEPLIBCOM_ERR)
 	$(E) "	LD $@"
 	$(Q) $(CC) -o tst_refcount $(srcdir)/ea_refcount.c \
-		$(ALL_CFLAGS) -DTEST_PROGRAM $(LIBCOM_ERR) $(LIBEXT2FS) 
+		$(ALL_CFLAGS) -DTEST_PROGRAM $(LIBCOM_ERR) $(LIBEXT2FS)
 
 tst_logfile: $(srcdir)/logfile.c
 	$(E) "	LD $@"
@@ -224,7 +224,7 @@
 	$(Q) for i in ext2 ext3 ext4 ext4dev; do \
 		$(ES) "	LINK $(root_sbindir)/fsck.$$i"; \
 		(cd $(DESTDIR)$(root_sbindir); \
-			$(LN) $(LINK_INSTALL_FLAGS) e2fsck fsck.$$i); \
+			$(LN_S) $(LINK_INSTALL_FLAGS) e2fsck fsck.$$i); \
 	done
 	$(Q) for i in $(MANPAGES); do \
 		for j in $(COMPRESS_EXT); do \
@@ -284,7 +284,7 @@
 	$(RM) -f .depend Makefile $(srcdir)/TAGS $(srcdir)/Makefile.in.old
 
 # +++ Dependency line eater +++
-# 
+#
 # Makefile dependencies follow.  This must be the last section in
 # the Makefile.in file
 #
diff -Naur a/misc/Makefile.in b/misc/Makefile.in
--- a/misc/Makefile.in	2016-01-20 19:48:17.032089859 +0100
+++ b/misc/Makefile.in	2016-01-20 19:49:51.224082142 +0100
@@ -86,16 +86,16 @@
 		$(srcdir)/ismounted.c $(srcdir)/../e2fsck/profile.c \
 		$(srcdir)/e2undo.c $(srcdir)/e2freefrag.c $(srcdir)/create_inode.c
 
-LIBS= $(LIBEXT2FS) $(LIBCOM_ERR) 
+LIBS= $(LIBEXT2FS) $(LIBCOM_ERR)
 DEPLIBS= $(LIBEXT2FS) $(DEPLIBCOM_ERR)
 PROFILED_LIBS= $(PROFILED_LIBEXT2FS) $(PROFILED_LIBCOM_ERR)
 PROFILED_DEPLIBS= $(PROFILED_LIBEXT2FS) $(DEPPROFILED_LIBCOM_ERR)
 
-STATIC_LIBS= $(STATIC_LIBEXT2FS) $(STATIC_LIBCOM_ERR) 
-STATIC_DEPLIBS= $(STATIC_LIBEXT2FS) $(DEPSTATIC_LIBCOM_ERR) 
+STATIC_LIBS= $(STATIC_LIBEXT2FS) $(STATIC_LIBCOM_ERR)
+STATIC_DEPLIBS= $(STATIC_LIBEXT2FS) $(DEPSTATIC_LIBCOM_ERR)
 
-LIBS_E2P= $(LIBE2P) $(LIBCOM_ERR) 
-DEPLIBS_E2P= $(LIBE2P) $(DEPLIBCOM_ERR) 
+LIBS_E2P= $(LIBE2P) $(LIBCOM_ERR)
+DEPLIBS_E2P= $(LIBE2P) $(DEPLIBCOM_ERR)
 
 COMPILE_ET=$(top_builddir)/lib/et/compile_et --build-tree
 
@@ -330,12 +330,12 @@
 
 filefrag: $(FILEFRAG_OBJS)
 	$(E) "	LD $@"
-	$(Q) $(CC) $(ALL_LDFLAGS) -o filefrag $(FILEFRAG_OBJS) 
+	$(Q) $(CC) $(ALL_LDFLAGS) -o filefrag $(FILEFRAG_OBJS)
 
 filefrag.profiled: $(FILEFRAG_OBJS)
 	$(E) "	LD $@"
 	$(Q) $(CC) $(ALL_LDFLAGS) -g -pg -o filefrag.profiled \
-		$(PROFILED_FILEFRAG_OBJS) 
+		$(PROFILED_FILEFRAG_OBJS)
 
 tst_ismounted: $(srcdir)/ismounted.c $(STATIC_LIBEXT2FS) $(DEPLIBCOM_ERR)
 	$(E) "	LD $@"
@@ -382,17 +382,17 @@
 	$(E) "	SUBST $@"
 	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/e4defrag.8.in e4defrag.8
 
-dumpe2fs.8: $(DEP_SUBSTITUTE) $(srcdir)/dumpe2fs.8.in 
+dumpe2fs.8: $(DEP_SUBSTITUTE) $(srcdir)/dumpe2fs.8.in
 	$(E) "	SUBST $@"
 	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/dumpe2fs.8.in dumpe2fs.8
 
 badblocks.8: $(DEP_SUBSTITUTE) $(srcdir)/badblocks.8.in
 	$(E) "	SUBST $@"
-	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/badblocks.8.in badblocks.8	
+	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/badblocks.8.in badblocks.8
 
 fsck.8: $(DEP_SUBSTITUTE) $(srcdir)/fsck.8.in
 	$(E) "	SUBST $@"
-	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/fsck.8.in fsck.8 
+	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/fsck.8.in fsck.8
 
 blkid.8: $(DEP_SUBSTITUTE) $(srcdir)/blkid.8.in
 	$(E) "	SUBST $@"
@@ -408,19 +408,19 @@
 
 chattr.1: $(DEP_SUBSTITUTE) $(srcdir)/chattr.1.in
 	$(E) "	SUBST $@"
-	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/chattr.1.in chattr.1 
+	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/chattr.1.in chattr.1
 
-lsattr.1: $(DEP_SUBSTITUTE) $(srcdir)/lsattr.1.in 
+lsattr.1: $(DEP_SUBSTITUTE) $(srcdir)/lsattr.1.in
 	$(E) "	SUBST $@"
-	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/lsattr.1.in lsattr.1 
+	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/lsattr.1.in lsattr.1
 
-uuidgen.1: $(DEP_SUBSTITUTE) $(srcdir)/uuidgen.1.in 
+uuidgen.1: $(DEP_SUBSTITUTE) $(srcdir)/uuidgen.1.in
 	$(E) "	SUBST $@"
-	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/uuidgen.1.in uuidgen.1 
+	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/uuidgen.1.in uuidgen.1
 
-blkid.1: $(DEP_SUBSTITUTE) $(srcdir)/blkid.1.in 
+blkid.1: $(DEP_SUBSTITUTE) $(srcdir)/blkid.1.in
 	$(E) "	SUBST $@"
-	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/blkid.1.in blkid.1 
+	$(Q) $(SUBSTITUTE_UPTIME) $(srcdir)/blkid.1.in blkid.1
 
 e2freefrag.8: $(DEP_SUBSTITUTE) $(srcdir)/e2freefrag.8.in
 	$(E) "	SUBST $@"
@@ -450,14 +450,14 @@
 	$(Q) for i in ext2 ext3 ext4 ext4dev; do \
 		$(ES) "	LINK $(root_sbindir)/mkfs.$$i"; \
 		(cd $(DESTDIR)$(root_sbindir); \
-			$(LN) $(LINK_INSTALL_FLAGS) mke2fs mkfs.$$i); \
+			$(LN_S) $(LINK_INSTALL_FLAGS) mke2fs mkfs.$$i); \
 	done
 	$(Q) (cd $(DESTDIR)$(root_sbindir); \
-		$(LN) $(LINK_INSTALL_FLAGS) tune2fs e2label)
+		$(LN_S) $(LINK_INSTALL_FLAGS) tune2fs e2label)
 	$(Q) if test -n "$(FINDFS_LINK)"; then \
 		$(ES) "	LINK $(root_sbindir)/findfs"; \
 		(cd $(DESTDIR)$(root_sbindir); \
-		$(LN) $(LINK_INSTALL_FLAGS) tune2fs $(FINDFS_LINK)); \
+		$(LN_S) $(LINK_INSTALL_FLAGS) tune2fs $(FINDFS_LINK)); \
 	fi
 	$(Q) for i in $(UPROGS); do \
 		$(ES) "	INSTALL $(bindir)/$$i"; \
@@ -602,7 +602,7 @@
 	$(RM) -f .depend Makefile $(srcdir)/TAGS $(srcdir)/Makefile.in.old
 
 # +++ Dependency line eater +++
-# 
+#
 # Makefile dependencies follow.  This must be the last section in
 # the Makefile.in file
 #
